### PR TITLE
FEATURE: Column order changing in Column configuration dialog (2023.1)

### PR DESCRIPTION
**Pull request by ptomask**

### CODE REVIEW

File: frontend-html/public/locale/localization_cs-CZ.json
Comment (tvavrda) [inline frontend-html/public/locale/localization_cs-CZ.json:1]: @washibana can you provide translations in other languages?

**Review by tvavrda — CHANGES_REQUESTED** [review]

File: frontend-html/public/locale/localization_cs-CZ.json
Comment (washibana) [reply frontend-html/public/locale/localization_cs-CZ.json:1]: Translations will be part of another PR.

File: frontend-html/src/gui/Components/Dialogs/ColumnsDialog.tsx [hunk 1]
```diff
@@ -47,39 +56,180 @@ export class ColumnsDialog extends React.Component<{
...
+    );
+  }
+  @observable temporaryPropertiesOrder: IProperty[] = [];
+
+  @action.bound handleOrderRowClick(id: any) {
+    this.selectedColumnId = id;
+  }
+
+  @action.bound handleOrderUp() {
+    const ord = this.temporaryPropertiesOrder;
```
Comment (washibana) [inline frontend-html/src/gui/Components/Dialogs/ColumnsDialog.tsx:97]: Please use full words in the names of varialbles. `order`, `selectedIndex`.

File: frontend-html/src/gui/Components/Dialogs/ColumnsDialog.tsx [hunk 2]
```diff
@@ -47,39 +56,180 @@ export class ColumnsDialog extends React.Component<{
...
+
+  @observable selectedColumnId: any = null;
+  @computed get selectedColumnIndex() {
+    return this.temporaryPropertiesOrder.findIndex(
+      (property) => property.id === this.selectedColumnId
+    );
+  }
+  @observable temporaryPropertiesOrder: IProperty[] = [];
+
+  @action.bound handleOrderRowClick(id: any) {
```
Comment (washibana) [inline frontend-html/src/gui/Components/Dialogs/ColumnsDialog.tsx:92]: It is better to know the meaning of `id`. E. g. `columnId`?

**Review by washibana — CHANGES_REQUESTED** [review]

**Review by washibana — APPROVED** [review]

**Review by washibana — APPROVED** [review]

**Review by tvavrda — APPROVED** [review]

**Pull request merged**